src/navsections/HelmChartSectionBlueprint/HelmChartQuickAction.tsx

```diff
@@ -8,50 +8,78 @@ import {ItemCustomComponentProps} from '@models/navigator';
 
 import {useAppDispatch, useAppSelector} from '@redux/hooks';
 import {selectHelmValuesFile} from '@redux/reducers/main';
+import {selectValuesFile} from '@redux/selectors';
 import {restartPreview, startPreview, stopPreview} from '@redux/services/preview';
 
+import {QuickActionCompare} from '@components/molecules/QuickActionCompare';
 import QuickActionPreview from '@components/molecules/QuickActionPreview';
 
 import {defineHotkey} from '@utils/defineHotkey';
+import {isDefined} from '@utils/filter';
 
 const QuickAction = (props: ItemCustomComponentProps) => {
   const {itemInstance} = props;
   const dispatch = useAppDispatch();
-  const selectedValuesFileId = useAppSelector(state => state.main.selectedValuesFileId);
-  const previewValuesFileId = useAppSelector(state => state.main.previewValuesFileId);
-
-  const isItemBeingPreviewed = useMemo(
-    () => previewValuesFileId !== undefined && previewValuesFileId === itemInstance.id,
-    [previewValuesFileId, itemInstance]
-  );
+  const thisValuesFile = useAppSelector(state => {
+    return selectValuesFile(state.main, itemInstance.id);
+  });
+  const selectedValuesFile = useAppSelector(state => {
+    return selectValuesFile(state.main, state.main.selectedPreviewConfigurationId);
+  });
+  const previewValuesFile = useAppSelector(state => {
+    return selectValuesFile(state.main, state.main.previewValuesFileId);
+  });
+  const isAnyPreviewing = isDefined(previewValuesFile);
+  const isThisPreviewing = itemInstance.id === previewValuesFile?.id;
+  const isThisSelected = itemInstance.id === selectedValuesFile?.id;
 
   const selectAndPreviewHelmValuesFile = useCallback(() => {
-    if (itemInstance.id !== selectedValuesFileId) {
+    if (!isThisSelected) {
       dispatch(selectHelmValuesFile({valuesFileId: itemInstance.id}));
     }
-    if (itemInstance.id !== previewValuesFileId) {
+    if (!isThisPreviewing) {
       startPreview(itemInstance.id, 'helm', dispatch);
     } else {
       stopPreview(dispatch);
     }
-  }, [itemInstance, selectedValuesFileId, previewValuesFileId, dispatch]);
+  }, [isThisSelected, isThisPreviewing, dispatch, itemInstance.id]);
 
   const reloadPreview = useCallback(() => {
-    if (itemInstance.id !== selectedValuesFileId) {
+    if (!isThisSelected) {
       dispatch(selectHelmValuesFile({valuesFileId: itemInstance.id}));
     }
 
     restartPreview(itemInstance.id, 'helm', dispatch);
-  }, [itemInstance, selectedValuesFileId, dispatch]);
+  }, [isThisSelected, itemInstance.id, dispatch]);
 
   useHotkeys(defineHotkey(hotkeys.RELOAD_PREVIEW.key), () => {
     reloadPreview();
   });
 
+  if (isAnyPreviewing && !isThisPreviewing) {
+    return (
+      <QuickActionCompare
```

Comment:
@devcatalin To make the feature more discoverable, I wanted to make the "Compare" show even while not hovering, but only on this specific quick action. On first sight it doesn't really seem possible with current blueprint though I quickly wanted to check with you to see if I'm overlooking something.